client/src/containers/PostDashboard.tsx

```diff
@@ -46,14 +57,46 @@ function RecentPosts() {
 }
 
 export default function PostDashboard() {
+  const [currentUser, setUser] = React.useState(null);
+  const [isLoading, setLoading] = React.useState(true);
+  const navigate = useNavigate();
+
+  const authBarrier = () => {
+    // force a login
+    navigate('/');
+    setUser(null);
+  };
```

Comment:
We could to something like this using react router, this way it automatically does checks on all pages.

Reply:
How do you mean? Like disabling the "/dashboard" route on App.tsx until they are authed?

Reply:
https://ui.dev/react-router-v5-protected-routes-authentication/

On loading a route you can check if the user object exists in app state. If not, redirect out to the login page, otherwise display the component for that page. 

```typescript
export function ProtectedRoute(props: {component: React.Component}) {
    const [user, ...] = getUserFromContext(); // some logic here
   if (user & user.isAuthenticated) {
     return <component/>
  }
  redirectToLoginPage(); // you get it
}
```

It's much safer to have it centralized in one location